aws/wafv2_helper.go

```diff
@@ -862,20 +1069,107 @@ func flattenWafv2RuleAction(a *wafv2.RuleAction) interface{} {
 	m := map[string]interface{}{}
 
 	if a.Allow != nil {
-		m["allow"] = make([]map[string]interface{}, 1)
+		m["allow"] = flattenWafv2Allow(a.Allow)
 	}
 
 	if a.Block != nil {
-		m["block"] = make([]map[string]interface{}, 1)
+		m["block"] = flattenWafv2Block(a.Block)
 	}
 
 	if a.Count != nil {
-		m["count"] = make([]map[string]interface{}, 1)
+		m["count"] = flattenWafv2Count(a.Count)
 	}
 
 	return []interface{}{m}
 }
 
+func flattenWafv2Allow(a *wafv2.AllowAction) interface{} {
+	if a == nil {
+		return map[string]interface{}{}
+	}
+	m := map[string]interface{}{}
+
+	if a.CustomRequestHandling != nil {
+		m["custom_request_handling"] = flattenWafv2CustomRequestHandling(a.CustomRequestHandling)
+	}
+
+	return []interface{}{m}
+}
+
+func flattenWafv2Block(a *wafv2.BlockAction) interface{} {
+	if a == nil {
+		return map[string]interface{}{}
+	}
+
+	m := map[string]interface{}{}
+
+	if a.CustomResponse != nil {
+		m["custom_response"] = flattenWafv2CustomResponse(a.CustomResponse)
+	}
+
+	return []interface{}{m}
+}
+
+func flattenWafv2Count(a *wafv2.CountAction) interface{} {
+	if a == nil {
+		return map[string]interface{}{}
+	}
+	m := map[string]interface{}{}
+
+	if a.CustomRequestHandling != nil {
+		m["custom_request_handling"] = flattenWafv2CustomRequestHandling(a.CustomRequestHandling)
+	}
+
+	return []interface{}{m}
+}
+
+func flattenWafv2CustomRequestHandling(c *wafv2.CustomRequestHandling) interface{} {
+	if c == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{
+		"insert_header": flattenWafv2CustomHeaders(c.InsertHeaders),
+	}
+
+	return []interface{}{m}
+}
+
+func flattenWafv2CustomResponse(r *wafv2.CustomResponse) interface{} {
+	if r == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{
+		"response_code":   int(aws.Int64Value(r.ResponseCode)),
+		"response_header": flattenWafv2CustomHeaders(r.ResponseHeaders),
+	}
+
+	return []interface{}{m}
+}
+
+func flattenWafv2CustomHeaders(h []*wafv2.CustomHTTPHeader) interface{} {
+	out := make([]interface{}, len(h))
+	for i, header := range h {
+		out[i] = flattenWafv2CustomHeader(header)
+	}
+
+	return out
+}
+
+func flattenWafv2CustomHeader(h *wafv2.CustomHTTPHeader) interface{} {
```

Comment:
let's set the return type to a map here since it's the only type possible
```suggestion
func flattenWafv2CustomHeader(h *wafv2.CustomHTTPHeader) map[string]interface{} {
```